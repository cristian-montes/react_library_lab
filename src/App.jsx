import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import Home from './views/Home/Home'
import BookList from './views/Books/BookList'
import BookDetail from './views/Books/BookDetail'
import Header from './components/header/Header'
import './App.css'

function App() {
  // TODO: Add routes to books & views
  return (
    <main className="container">
      <Router>
        <Header />
        <Switch>
          <Route path="/" exact component={Home} />
          <Route path="/books" exact component={BookList} />
          <Route path="/books/:id" component={BookDetail} />
          <Home />
          <BookList />
        </Switch>
      </Router>
    </main>
  )
}

export default App
