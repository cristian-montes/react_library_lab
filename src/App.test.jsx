import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import App from './App'

// eslint-disable-next-line no-undef
it('renders a welcome message when signed out', async () => {
  const { container } = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  )

  const texto = screen.getAllByText(/library catalog/i)

  // eslint-disable-next-line no-undef
  // expect(texto).toBeInTheDocument()

  // eslint-disable-next-line no-undef
  expect(container).toMatchSnapshot()
})
