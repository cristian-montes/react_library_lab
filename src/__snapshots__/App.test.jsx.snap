// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a welcome message when signed out 1`] = `
<div>
  <main
    class="container"
  >
    <div>
      <h1>
         Library Catalog 
      </h1>
    </div>
    <div>
      <h2>
        Welcome to Dangalf's Library of Computer Science!
      </h2>
      <p>
        Check out the amazing selection of books in our
         
        <a
          class="linkie"
          href="/books"
        >
          catalog
        </a>
      </p>
    </div>
  </main>
</div>
`;
